airbyte-cdk/python/airbyte_cdk/sources/declarative/interpolation/macros.py

```diff
@@ -95,7 +97,7 @@ def day_delta(num_days: int, format: str = "%Y-%m-%dT%H:%M:%S.%f%z") -> str:
     return (datetime.datetime.now(datetime.timezone.utc) + datetime.timedelta(days=num_days)).strftime(format)
 
 
-def duration(datestring: str) -> datetime.timedelta:
+def duration(datestring: str) -> Union[datetime.timedelta, isodate.Duration]:
```

Comment:
why can this method return both type? Is there any way we can restrict the return type to only one of the two?

Reply:
> why can this method return both type?

because of the `parent` method :

![image](https://github.com/airbytehq/airbyte/assets/36314070/e5d0c13d-73e7-4a1d-b272-aaf32547e186)

We have not met it because any duration we use starts with 'P' and therefore matches `line 87`.
I cannot remove `Duration` from hint because mypy will raise an error.